src/manual.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2017 Chris Kowalczyk <ckowalczyk@suse.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "manual.h"
+
+#include "transport.h"
+#include "ticket.h"
+#include "config.h"
+#include "log.h"
+#include "request.h"
+
+
+int manual_selection(struct ticket_config *tk,
+	struct booth_site *preference, int update_term, cmd_reason_t reason)
+{
+	if (local->type != SITE)
+		return 0;
+
+	tk_log_debug("starting manual selection");
+	tk_log_debug("selection caused by %s %s",
+				state_to_string(reason),
+				reason == OR_AGAIN ? state_to_string(tk->election_reason) : "" );
+
+	// Manual selection is done without any delay, the leader is assigned
+	set_leader(tk, local);
+	set_state(tk, ST_LEADER);
+
+	// Manual tickets never expire, we don't specify expiration time
+
+	// Make sure that election_end field is empty
+	time_reset(&tk->election_end);
+
+	if (is_time_set(&tk->delay_commit) && all_sites_replied(tk)) {
```

Comment:
Was there anything broadcasted before so that we should expect replies?

Reply:
hm, acks for heartbits? in the unlikely case when ticket was already granted.. I agree, not a very likely or important scenario.. Can remove if you prefer

Reply:
Can we then check only in that case? Or are the acks important at all?

Reply:
After some thinking, perhaps for manual tickets we should always reset __delay_commit__ (that is, say, to not set __delay_commit__ at all for manual tickets).  After all, we don't care for the responses from other nodes, a manual ticket is granted immediately after the command.

Reply:
In the case of granting a granted ticket, is it probably supposed to receive certain reply from the previous holder about a successful/failed revocation so that it knows whether it's safe to grant now?

Reply:
Of course it probably doesn't matter if re-grant is not the combination of revoke&grant ;)

Reply:
Granting a granted ticket doesn't trigger another leader to send any kind of confirmation.
So, I guess we don't have to wait for any messages here (as I mentioned a few comments above). I will remove this IF and always clear tk->delay_commit time